### ISSUE TITLE
String labels for xtic

### ISSUE THREAD

**Issue by bonorumetmalorum** [SiegeLord/RustGnuplot#35]
I was creating a bar chart using this library but found that it would not allow for strings to be attached to xtics. If this could be added it would be immensely useful for creating bar charts.

**Comment by SiegeLord** [comment]
You can do this:

```rust
let mut fg = Figure::new();
c.set_term(&mut fg);

fg.axes2d()
    .boxes(&[1., 2., 3.], &[1., 2., 3.], &[Color("gray"), BorderColor("black")])
    .set_x_ticks_custom(
        vec![
            Major(1. as f32, Fix("A".into())),
            Major(2. as f32, Fix("B".into())),
            Major(3. as f32, Fix("C".into()))
        ],
        &[],
        &[],
    )
    .set_y_range(Fix(0.0), Auto);
fg.show();
```
to get:

![image](https://user-images.githubusercontent.com/480550/53290483-34440200-3759-11e9-9b6f-2c4f695c2d39.png)

**Comment by SiegeLord** [comment]
Or, if your have an array of labels:

```
let mut fg = Figure::new();
c.set_term(&mut fg);

let x = [1., 2., 3.];
let y = [1., 2., 3.];
let l = ["A", "B", "C"];

fg.axes2d()
    .boxes(&x, &y, &[Color("gray"), BorderColor("black")])
    .set_x_ticks_custom(
        x.iter().zip(&l).map(|(&x, &l)| Major(x as f32, Fix(l.into()))),
        &[],
        &[],
    )
    .set_y_range(Fix(0.0), Auto);
fg.show();
```

**Comment by bonorumetmalorum** [comment]
Thank you! this helped a lot :+1:
